oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/config/group/EndpointNameGrouping.java

```diff
@@ -99,18 +105,30 @@ public Tuple2<String, Boolean> format(String serviceName, String endpointName) {
                     // Algorithm side should not return a pattern that has no {var} in it else this
                     // code may accidentally retrieve the size 1 queue created by unformatted endpoint
                     // The queue size is 10, which means only cache the first 10 formatted names.
-                    final ArrayBlockingQueue<String> formattedURIs = svrHttpUris.computeIfAbsent(
+                    final Queue<String> formattedURIs = svrHttpUris.computeIfAbsent(
```

Comment:
If we keep this as a queue, we can't be sure the URI is distinct. The first N element would fill the queue for sure.

Reply:
Current version is AI training, we don't have concern about the duplicate URIs.